src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -154,17 +155,28 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
                 // Only actual dictionaries are supported, as arbitrary types
                 // can implement multiple IDictionary interfaces and thus introduce
                 // multiple different interpretations.
+                var valueAsList = enumerable.Cast<object>();
```

Comment:
All LINQ operators can potentially allocate an `IEnumerable<T>`, even without it ever being enumerated. This is why performance-critical code like Roslyn etc. doesn't allow use of LINQ :smile: